lib/modules/manager/devbox/extract.spec.ts

```diff
@@ -0,0 +1,292 @@
+import { codeBlock } from 'common-tags';
+import { extractPackageFile } from '.';
+
+describe('modules/manager/devbox/extract', () => {
+  describe('extractPackageFile', () => {
+    it('returns null when the devbox JSON file is empty', () => {
+      const result = extractPackageFile('', 'devbox.lock');
+      expect(result).toBeNull();
+    });
+
+    it('returns null when the devbox JSON file is malformed', () => {
+      const result = extractPackageFile('malformed json}}}}}', 'devbox.lock');
+      expect(result).toBeNull();
+    });
+
+    it('returns null when the devbox JSON file has no packages', () => {
+      const result = extractPackageFile('{}', 'devbox.lock');
+      expect(result).toBeNull();
+    });
+
+    it('returns a package dependency when the devbox JSON file has a single package', () => {
+      const result = extractPackageFile(
+        codeBlock`
+        {
+          "packages": ["nodejs@20.1.8"]
+        }
+      `,
+        'devbox.lock',
+      );
+      expect(result).toEqual({
+        deps: [
+          {
+            depName: 'nodejs',
+            currentValue: '20.1.8',
+            datasource: 'devbox',
+            packageName: 'nodejs',
+          },
+        ],
+      });
+    });
+
+    it('returns a package dependency when the devbox JSON file has a single package with a version object', () => {
+      const result = extractPackageFile(
+        codeBlock`
+        {
+          "packages": {
+            "nodejs": "20.1.8"
+          }
+        }
+      `,
+        'devbox.lock',
+      );
+      expect(result).toEqual({
+        deps: [
+          {
+            depName: 'nodejs',
+            currentValue: '20.1.8',
+            datasource: 'devbox',
+            packageName: 'nodejs',
+          },
+        ],
+      });
+    });
+
+    it('returns null when the devbox JSON file has a single package with a version object and a version range', () => {
+      const result = extractPackageFile(
+        codeBlock`
+        {
+          "packages": {
+            "nodejs": "^20.1.8"
+          }
+        }
+      `,
+        'devbox.lock',
+      );
+      expect(result).toEqual({
+        deps: [
+          {
+            currentValue: '^20.1.8',
+            datasource: 'devbox',
+            depName: 'nodejs',
+            packageName: 'nodejs',
+            skipReason: 'invalid-version',
```

Comment:
why is this invalid?

Reply:
version ranges are not allowed in devbox, so it's the ^ which is causing that to be an invalid version